src/foam/nanos/notification/email/ApplyBaseArgumentsEmailPropertyService.js

```diff
@@ -38,75 +38,79 @@ foam.CLASS({
         Logger logger = Loggers.logger(x, this);
 
         User user = (User) emailMessage.findUser(getX());
-        x = foam.util.Auth.sudo(x, user);
+        if ( user != null ) {
+          x = foam.util.Auth.sudo(x, user);
+          emailMessage.setSpid(user.getSpid());
+          String sendTo = (String) templateArgs.get("sendTo");
+          if ( SafetyUtil.isEmpty(sendTo) )
+            templateArgs.put("sendTo", user.getEmail());
+          templateArgs.put("locale", user.getLanguage().getCode().toString());
+        }
 
-        emailMessage.setSpid(user.getSpid());
+        // AppConfig config
+        AppConfig appConfig = (AppConfig) x.get("appConfig");
```

Comment:
This needs to be done in a user context, so the Auth.sudo context above.  Else you get the default for the system which isn't very useful. 
If there is no user then at least the spid is required. 
Themes believe has logic for this. 

So, if user null, then at least set the spid in the context and first get the Theme, then the appconfig.